components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal

```diff
@@ -112,29 +114,58 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
-                    path: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PATH,
-                    "${ballerina.home}/bre/security/ballerinaTruststore.p12"),
-                    password: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PASSWORD, "ballerina")
-                },
-                verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
-            }
+    http:ClientSecureSocket secureSocket = {
+        trustStore: {
+            path: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PATH,
+            "${ballerina.home}/bre/security/ballerinaTruststore.p12"),
+            password: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PASSWORD, "ballerina")
+        },
+        verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
+    };
+    http:OutboundAuthConfig? auth = ();
+    if (getConfigBooleanValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, ENABLED, true)) {
+        auth:OutboundBasicAuthProvider basicAuthOutboundProvider = new({
+            username: getConfigValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, USERNAME, "admin"),
+            password: getConfigValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, PASSWORD, "admin")
+        });
+        http:BasicAuthHandler basicAuthOutboundHandler = new(basicAuthOutboundProvider);
+        auth = {authHandler: basicAuthOutboundHandler};
+    } else if (getConfigBooleanValue(KM_CONF_SECURITY_OAUTH2_INSTANCE_ID, ENABLED, false)) {
+        oauth2:OutboundOAuth2Provider|error oauth2Provider = getOauth2OutboundProvider();
+        if (oauth2Provider is oauth2:OutboundOAuth2Provider) {
+            http:BearerAuthHandler bearerAuthOutboundHandler = new(oauth2Provider);
+            auth = {authHandler: bearerAuthOutboundHandler};
+        } else {
+            printFullError(KEY_GW_LISTNER, oauth2Provider);
         }
+    } else {
+        printWarn(KEY_GW_LISTNER, "Key validation service security confogurations not enabled.");
+    }
+    http:ClientConfiguration clientConfig = {
+        auth: auth,
+        cache: {enabled: false},
+        secureSocket: secureSocket
     };
-    OAuth2KeyValidationProvider oauth2KeyValidationProvider = new (keyValidationServerConfig);
+    oauth2:IntrospectionServerConfig  keyValidationConfig = {
+        url: getConfigValue(KM_CONF_INSTANCE_ID, KM_SERVER_URL, "https://localhost:9443"),
+        clientConfig: clientConfig
+    };
+    string introspectURL = getConfigValue(KM_CONF_INSTANCE_ID, KM_SERVER_URL, "https://localhost:9443");
+    string keymanagerContext = getConfigValue(KM_CONF_INSTANCE_ID, KM_TOKEN_CONTEXT, "oauth2");
+    introspectURL = (introspectURL.endsWith(PATH_SEPERATOR)) ? introspectURL + keymanagerContext : introspectURL + PATH_SEPERATOR + keymanagerContext;
+    introspectURL = (introspectURL.endsWith(PATH_SEPERATOR)) ? introspectURL + INTROSPECT_CONTEXT : introspectURL + PATH_SEPERATOR + INTROSPECT_CONTEXT;
```

Comment:
`introspectURL` is going to be the result of 157 anyway. Result of 156 is not used. Mistake?

Reply:
Result of 156 is used in line 157

Reply:
Oops my mistake